src/compute/src/render/continual_task.rs

```diff
@@ -261,7 +261,7 @@ where
         let sink_write_frontier = Rc::new(RefCell::new(Antichain::from_elem(Timestamp::minimum())));
         collection.sink_write_frontier = Some(Rc::clone(&sink_write_frontier));
 
-        // TODO(ct): Obey `compute_state.read_only_rx`
+        // TODO(ct2): Obey `compute_state.read_only_rx`
```

Comment:
Can you elaborate on why this isn't needed for M1? Without this, we can have two CTs from two different envd versions writing to the same shard at the same time, right? Is the thinking that that doesn't really matter for truncating system collections because we are just deleting records and that should always work the same between version updates?

Reply:
> MVs care about having a single cutover point because of the potential for thrashing between two answers in the self-correcting sink, but CTs don't have that.

I wonder if that can't also happen with CTs. If the CT has the output as one of its inputs it will observe changes made by other replicas and might try to revert them, depending on its definition. Though I can't come up with a practical scenario/CT definition where this would actually happen.

My concern would be more around a "read-only" cluster making changes to durable state that can be observed by the existing environment. That feels wrong, even though it perhaps doesn't matter for 0dt.

Anyway, I'm okay with punting this to M2.

Reply:
> > MVs care about having a single cutover point because of the potential for thrashing between two answers in the self-correcting sink, but CTs don't have that.
> 
> I wonder if that can't also happen with CTs. If the CT has the output as one of its inputs it will observe changes made by other replicas and might try to revert them, depending on its definition. Though I can't come up with a practical scenario/CT definition where this would actually happen.

Ah good call. Yeah, I'm convinced.

> My concern would be more around a "read-only" cluster making changes to durable state that can be observed by the existing environment. That feels wrong, even though it perhaps doesn't matter for 0dt.

I realize that there are two things you could call M1: one is having one of these builtin CTs running in the background, and another is using it to power anything user facing. Given that the latter will require a 30d backfill like we're doing for the new `_history` collections, in practice this should be done by the time a user could observe it.

> Anyway, I'm okay with punting this to M2.

I think you've convinced me it should be M1. Given that I have to resolve a merge conflict anyway, I'll go ahead and switch it :joy: